apps/roam/src/utils/setBlockProps.ts

```diff
@@ -0,0 +1,48 @@
+import getBlockProps, {
+  type json,
+  getRawBlockProps,
+  normalizeProps,
+} from "./getBlockProps";
+import getPageUidByPageTitle from "roamjs-components/queries/getPageUidByPageTitle";
+import { DISCOURSE_CONFIG_PAGE_TITLE } from "~/utils/renderNodeConfigPage";
+
+export const deNormalizeProps = (props: json): json =>
```

Comment:
Can you explain why `deNormalizeProps` is required?  I believe we have been setting canvas without this for some time: https://github.com/DiscourseGraphs/discourse-graph/blob/ENG-418-create-and-store-roam-space-password/apps/roam/src/utils/useRoamStore.ts#L73-L99

Reply:
Ah, good to know, was not aware of that. My understanding was that the initial colon was something that roam expects, to be able to do datalog queries and computations. Some values may not fit there, and password may well be an example. But my understanding of props is still limited.

What I will do now: I'll keep the utility function as-is, but I'll add a boolean parameter and make it optional, with default false, and callers can ask to denormalize if they want the new values to be visible from datalog.